operator/pkg/client/roles/client.go

```diff
@@ -117,105 +114,89 @@ func (c *Client) Delete(ctx context.Context, role *redpandav1alpha2.RedpandaRole
+	if len(desiredPrincipals) == 0 && len(previouslyManaged) == 0 {
+		// No principals defined in Role or RoleBindings and nothing previously managed - don't touch membership
+		// Return empty slice to indicate manual management mode
+		return []string{}, nil
 	}
 
-	for i, principal := range toRemove {
-		if principal == "" {
-			return fmt.Errorf("principal at index %d is empty", i)
-		}
-		// Parse principal to extract username
-		name := parsePrincipal(principal)
-		removeMembers[i] = rpadmin.RoleMember{
-			Name:          name,
-			PrincipalType: principalTypeUser,
-		}
-	}
+	// Calculate members to add and remove
+	// Only remove principals WE previously managed (protects manual additions)
+	toAdd, toRemove := calculateMembershipChanges(currentPrincipals, desiredPrincipals, previouslyManaged)
 
-	// Use bulk update for efficiency
-	if len(addMembers) > 0 || len(removeMembers) > 0 {
-		_, err := c.adminClient.UpdateRoleMembership(ctx, roleName, addMembers, removeMembers, false)
+	// Update membership if there are changes
+	if len(toAdd) > 0 || len(toRemove) > 0 {
+		_, err := c.adminClient.UpdateRoleMembership(ctx, role.Name, toAdd, toRemove, false)
 		if err != nil {
-			return fmt.Errorf("updating role membership for role %s: %w", roleName, err)
+			return nil, fmt.Errorf("updating role membership for %s: %w", role.Name, err)
 		}
 	}
 
-	return nil
+	// Return the desired principals we successfully synced
+	return desiredPrincipals, nil
 }
 
-// calculateMembershipChanges determines which principals to add and remove
-func calculateMembershipChanges(current, desired []string) (toAdd, toRemove []string) {
+// calculateMembershipChanges determines which principals to add and remove.
+// It only removes principals that were previously managed by the operator (in previouslyManaged).
+// This protects manually added principals from being removed.
+func calculateMembershipChanges(current, desired, previouslyManaged []string) (toAdd, toRemove []rpadmin.RoleMember) {
 	// Find principals to add (in desired but not in current)
 	for _, principal := range desired {
 		if !slices.Contains(current, principal) {
-			toAdd = append(toAdd, principal)
+			principalType, name := parsePrincipal(principal)
+			toAdd = append(toAdd, rpadmin.RoleMember{
+				Name:          name,
+				PrincipalType: principalType,
+			})
 		}
 	}
 
-	// Find principals to remove (in current but not in desired)
-	for _, principal := range current {
+	// Only remove principals that WE previously managed AND are no longer desired
+	// This protects manually added principals (not in previouslyManaged)
+	for _, principal := range previouslyManaged {
 		if !slices.Contains(desired, principal) {
-			toRemove = append(toRemove, principal)
+			// We managed it before, but it's no longer desired - remove it
+			principalType, name := parsePrincipal(principal)
+			toRemove = append(toRemove, rpadmin.RoleMember{
+				Name:          name,
+				PrincipalType: principalType,
+			})
```

Comment:
NIT: Not a blocker, but this check against previous RoleStatus principals does not fully fill declarative model. At least in my mind if user would like to manage something using declerative Role custom resource, anything will converge to the state declared in the spec.